src/net.h

```diff
@@ -314,15 +316,17 @@ class CConnman
 
 private:
     struct ListenSocket {
+    private:
+        NetPermissionFlags m_permissions;
+    public:
         SOCKET socket;
-        bool whitelisted;
-
-        ListenSocket(SOCKET socket_, bool whitelisted_) : socket(socket_), whitelisted(whitelisted_) {}
+        inline void AddSocketPermissionFlags(NetPermissionFlags& flags) const { NetPermissions::AddFlag(flags, m_permissions); }
+        ListenSocket(SOCKET socket_, NetPermissionFlags permissions_) : socket(socket_), m_permissions(permissions_) {}
```

Comment:
This gives me compiler warnings ("warning: ‘CConnman::ListenSocket::socket’ will be initialized after [-Wreorder] SOCKET socket;"), suggest to change the order somewhere (in the constructor or in the struct).